### PR TITLE
api/store:set state do not support set from or to Tombstone anymore

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: shirly <AndreMouche@126.com>

### What problem does this PR solve?

To close issue https://github.com/tikv/pd/issues/3076

This PR does not support setting a store's state to or from `Tombstone` by the API `set store state`

### What is changed and how it works?

- the API to update the store's state does not support `Tombstone` anymore:
```
POST /store/{id}/state?state=${state}
```
state query string true "state" Enums(Up, Offline), `Tombstone` is not supported anymore. Also, this PR does not support to up a store from `tombstone` or `physically destroyed`

### Check List

Tests

- Unit test

Related changes

- https://github.com/pingcap/kvproto/pull/711
- https://github.com/tikv/pd/pull/3388

### Release note 
- No release note

### CODE REVIEW

File: server/core/store_option.go [hunk 1]
```diff
@@ -73,21 +73,30 @@ func SetStoreDeployPath(deployPath string) StoreCreateOption {
...
+		meta.State = metapb.StoreState_Up
+		store.meta = meta
+	}
+}
+
+// TombstoneStore set a store to tombstone.
+func TombstoneStore() StoreCreateOption {
+	return func(store *StoreInfo) {
+		meta := proto.Clone(store.meta).(*metapb.Store)
+		meta.State = metapb.StoreState_Tombstone
```
Comment (HunDunDM) [inline server/core/store_option.go:99]: Should `PhysicallyDestroyed` be changed to `false`? And maybe we should use `func setStoreState(state metapb.StoreState, physicallyDestroyed bool) StoreCreateOption` to reduce redundant code.

**Review by HunDunDM — COMMENTED** [review]
rest LGTM

File: server/api/store.go [hunk 2]
```diff
@@ -222,10 +222,18 @@ func (h *storeHandler) SetState(w http.ResponseWriter, r *http.Request) {
 		h.rd.JSON(w, http.StatusBadRequest, "invalid state")
 		return
 	}
+	var err error
+	switch metapb.StoreState(state) {
```
Comment (Yisaer) [inline server/api/store.go:226]: If the `state` is `tombstone`, I think we can return the response that `tombstone` is not supported to be set anymore.

File: server/core/store_option.go (hunk 1 above)
Comment (Yisaer) [reply server/core/store_option.go:99]: Agreed, the api design lgtm while we can use `setStoreState` to reduce the duplicated code.

**Review by Yisaer — COMMENTED** [review]
Rest LGTM

File: server/core/store_option.go (hunk 1 above)
Comment (AndreMouche) [reply server/core/store_option.go:99]: As PR #3388 discussed, SetStoreState are replaced by following `StoreCreateOption`s:
```
- UpStore() 
- OfflineStore(physicallyDestroyed bool)
- TombstoneStore()
```
which have clearer definition and logic for each `StoreCreateOption`

**Review by HunDunDM — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]
LGTM

**Review by nolouch — COMMENTED** [review]
rest LGTM.

**Pull request merged**